receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -129,6 +188,35 @@ func parseMessageToMetric(line string) (*statsDMetric, error) {
 			return nil, fmt.Errorf("unrecognized message part: %s", part)
 		}
 	}
+
+	switch result.statsdMetricType {
+	case "g":
+		f, err := strconv.ParseFloat(result.value, 64)
+		if err != nil {
+			return nil, fmt.Errorf("gauge: parse metric value string: %s", result.value)
+		}
+		result.floatvalue = f
+		result.metricType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+	case "c":
+		f, err := strconv.ParseFloat(result.value, 64)
+		if err != nil {
+			return nil, fmt.Errorf("counter: parse metric value string: %s", result.value)
+		}
+		i := int64(f)
+		if 0 < result.sampleRate && result.sampleRate < 1 {
+			i = int64(f / result.sampleRate)
+		}
+		result.intvalue = i
+		result.metricType = metricspb.MetricDescriptor_GAUGE_INT64
+	}
+
+	var tg []string
+	for n, k := range result.labelKeys {
+		if result.labelValues[n].HasValue {
+			tg = append(tg, k.Key+"="+result.labelValues[n].Value)
+		}
+	}
+	result.hash = result.name + strings.Join(tg, "")
```

Comment:
The OTel-Go SDK has provided a good solution to this problem, starting with a type called `label.Set`, which can return a `Distinct` value that is usable as a map key. It handles sorting and de-duplication, but the tricky part is to use an array of key-values so that no hashing is required. Can we import that library?

Then you can have a "description" for the code here like 

```
struct {
  Name string
  Type string // "c" or "g", etc.
  label.Distinct. // the label key:values
}
```